File: copy_blog_images.js
// javascript
const fs = require('fs-extra');
const path = require('path');

const sourceDir = 'source';
const targetDir = path.join(sourceDir, 'img');

async function copyImages(sourceDir, targetDir) {
  /**
   * 查找 sourceDir 下所有子目录中的 img 文件夹，将其中的图片复制到 targetDir，重复则覆盖。
   */
  async function walk(dir) {
    const files = fs.readdirSync(dir);

    for (const file of files) {
      const filePath = path.join(dir, file);
      const stat = fs.statSync(filePath);

      if (stat.isDirectory()) {
        if (filePath.endsWith('img') && filePath !== targetDir) {
          const imageFiles = fs.readdirSync(filePath);
          for (const imageFile of imageFiles) {
            const sourceFile = path.join(filePath, imageFile);
            const targetFile = path.join(targetDir, imageFile);
            try {
              await fs.copy(sourceFile, targetFile, { overwrite: true }); // copy 保留文件元数据
              console.log(`已复制: ${sourceFile} -> ${targetFile}`);
            } catch (e) {
              console.log(`复制失败: ${sourceFile} -> ${targetFile}, 错误: ${e}`);
            }
          }
        } else {
          await walk(filePath);
        }
      }
    }
  }

  await walk(sourceDir);
}

async function main() {
  // 确保目标文件夹存在
  fs.ensureDirSync(targetDir);
  await copyImages(sourceDir, targetDir);
  console.log('图片复制完成！');
}

main();

File: copy_blog_images.py
import os
import shutil

source_dir = "source"
target_dir = os.path.join(source_dir, "img")

def copy_images(source_dir, target_dir):
    """
    查找 source_dir 下所有子目录中的 img 文件夹，将其中的图片复制到 target_dir，重复则覆盖。
    """
    for root, dirs, files in os.walk(source_dir):
        if root.endswith("img") and root != target_dir:
            for file in files:
                source_file = os.path.join(root, file)
                target_file = os.path.join(target_dir, file)
                try:
                    shutil.copy2(source_file, target_file)  # copy2 保留文件元数据
                    print(f"已复制: {source_file} -> {target_file}")
                except Exception as e:
                    print(f"复制失败: {source_file} -> {target_file}, 错误: {e}")

if __name__ == "__main__":
    # 确保目标文件夹存在
    os.makedirs(target_dir, exist_ok=True)
    copy_images(source_dir, target_dir)
    print("图片复制完成！")